lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -82,6 +78,50 @@ def reset(cls):
         cls.set_values = {}
 
 
+class FakeIndexMap:
+    index_map: dict[str, IndexMapType] = {}
+
+    @staticmethod
+    def exists(dataset: FakeDataset) -> bool:
+        return bool(__class__.index_map)
+
+    @staticmethod
+    def find(dataset: FakeDataset, index: str) -> dict[str, list[str]]:
+        return {
+            i: d
+            for i, d in __class__.index_map.get(dataset.name, {}).items()
+            if i.startswith(index)
+        }
+
+    @staticmethod
+    def create(dataset: FakeDataset, map: JSONOBJECT) -> JSONOBJECT:
+        __class__.index_map[dataset.name] = map
+        return map
+
+    def add(self):
+        pass
```

Comment:
Should this function be named `commit()`?  (Currently, this function is never called -- if you aren't going to rename it, should it be removed?)